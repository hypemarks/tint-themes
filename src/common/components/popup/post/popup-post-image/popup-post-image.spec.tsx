import { render } from '@testing-library/react';
import React from 'react';
import { buildPost } from '../../../../../../test/generate';
import PopupPostImage from './popup-post-image';
import { IPostWithCTA } from '../../../../../index';

jest.mock('@tintup/tint-sdk/lib', () => {
  return {
    FileLinkClientService: jest.fn(() => ({
      noMetadata: jest.fn(() => ({
        resize: jest.fn(() => ({
          applyEffect: jest.fn(() => ({
            autoImage: jest.fn(() => ({
              compress: jest.fn(() => ({
                toString: jest.fn(),
              })),
            })),
          })),
        })),
      })),
    })),
  };
});

describe('Test <PopupPostImage /> Component', () => {
  const post = buildPost({ attributes: { image_url: 'image_url' } });
  const config = {
    onCtaClick: jest.fn,
    ctaConfig: { appendUtmToCta: {} },
    highlightConfig: {},
    isMounted: false,
    waterfallToggle: false,
  };

  it('should render snapshot', () => {
    const wrapper = render(
      <PopupPostImage
        post={post as IPostWithCTA}
        highlightConfig={config.highlightConfig}
        isMounted={config.isMounted}
        waterfallToggle={config.waterfallToggle}
        onVideoEnded={jest.fn}
        taggedProducts={[]}
      />
    );
    expect(wrapper.asFragment()).toMatchSnapshot();
  });

  it('should not render image', () => {
    const wrapper = render(
      <PopupPostImage
        post={post as IPostWithCTA}
        highlightConfig={config.highlightConfig}
        isMounted={config.isMounted}
        noImages={true}
        waterfallToggle={config.waterfallToggle}
        onVideoEnded={jest.fn}
        taggedProducts={[]}
      />
    );
    const image = wrapper.queryByTestId('image');
    expect(image).not.toBeInTheDocument();
  });
});
