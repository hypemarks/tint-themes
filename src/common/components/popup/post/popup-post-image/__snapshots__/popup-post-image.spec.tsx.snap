// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <PopupPostImage /> Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="popup-post-image"
    style="background: rgb(0, 0, 0);"
  >
    <div
      class="popup-post-image__image"
      data-testid="image"
    >
      <div
        class="popup-post-image__wrapper"
      >
        <div
          class="image popup-post-image__image image--normal image--loading"
          data-testid="image"
        >
          <div
            class="image-popup image-popup--horizontal"
          >
            <img
              alt=""
              src="image_url"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
